rest/data/pod.go

```diff
@@ -0,0 +1,42 @@
+package data
+
+import (
+	"github.com/evergreen-ci/evergreen/model/pod"
+	"github.com/pkg/errors"
+)
+
+// DBPodConnector implements the pod-related methods from the connector via
+// interactions with the database.
+type DBPodConnector struct{}
+
+// CheckPodSecret checks for a pod with a matching ID and secret in the
+// database.
+func (c *DBPodConnector) CheckPodSecret(id, secret string) error {
+	p, err := pod.FindOneByID(id)
+	if err != nil {
+		return err
+	}
+	if p == nil {
+		return errors.New("pod does not exist")
+	}
+	if secret != p.Secret {
+		return errors.New("pod secrets do not match")
```

Comment:
Nit: can this say something like `incorrect pod secret`?

Reply:
Done.